.github/workflows/go-lint.yml

```diff
@@ -3,37 +3,72 @@ on:
   push:
     branches:
       - master
-    paths-ignore:
-      - 'docs/**'
-      - 'web/**'
+    paths:
+      - 'api/**'
   pull_request:
     branches:
       - master
-    paths-ignore:
-      - 'docs/**'
-      - 'web/**'
+    paths:
+      - 'api/**'
 
 jobs:
+  go-filter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodule: true
+
+      - uses: ./.github/actions/paths-filter
+        id: changes
+        with:
+          filters: |
+            go:
+              - '**.go'
+          working-directory: 'api'
+          list-files: shell
+    outputs:
+      matches: ${{ steps.changes.outputs.go }}
+      files: ${{ steps.changes.outputs.go_files }}
+
   golangci:
     runs-on: ubuntu-latest
+    needs: go-filter
+    if: needs.go-filter.outputs.matches == 'true'
     steps:
       - uses: actions/checkout@v2
 
-      - name: run lint
-        run: make go-lint
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: latest
+          working-directory: api
+          args: --tests=false
+          only-new-issues: true
 
   gofmt:
     runs-on: ubuntu-latest
+    needs: go-filter
+    if: needs.go-filter.outputs.matches == 'true'
     steps:
       - uses: actions/checkout@v2
       - name: setup go
         uses: actions/setup-go@v1
         with:
           go-version: '1.13'
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
```

Comment:
Just want to know what's the use of this stage?

Reply:
This stage caches go mod and build to speed up go, see https://github.com/actions/cache/blob/main/examples.md#go---modules

Reply:
Pretty good !